app/src/main/java/org/dhis2/data/dhislogic/DhisEventUtils.kt

```diff
@@ -8,13 +8,13 @@ class DhisEventUtils @Inject constructor(val d2: D2) {
     fun newEventNeedsExtraInfo(eventUid: String): Boolean {
         val event = d2.eventModule().events().uid(eventUid)
             .blockingGet()
-        val stage = d2.programModule().programStages().uid(event.programStage())
+        val stage = d2.programModule().programStages().uid(event?.programStage())
             .blockingGet()
-        val program = d2.programModule().programs().uid(stage.program()?.uid())
+        val program = d2.programModule().programs().uid(stage?.program()?.uid())
             .blockingGet()
-        val hasCoordinates = stage.featureType() != null && stage.featureType() != FeatureType.NONE
+        val hasCoordinates = stage?.featureType() != null && stage.featureType() != FeatureType.NONE
         val hasNonDefaultCatCombo = d2.categoryModule().categoryCombos()
-            .uid(program.categoryComboUid()).blockingGet().isDefault != true
+            .uid(program?.categoryComboUid()).blockingGet()?.isDefault != true
```

Comment:
is there a possibility to call `d2.categoryModule().categoryCombos()uid(**null**).blockingGet` ? . Are null parameters allowed?